source/solvers/heat_transfer.cc

```diff
@@ -178,6 +178,106 @@ HeatTransfer<dim>::assemble_nitsche_heat_restriction(bool assemble_matrix)
     }
 }
 
+template <int dim>
+void
+HeatTransfer<dim>::postprocess_heat_flux_on_nitsche_ib()
+{
+  auto solids = *this->multiphysics->get_solids(
+    this->simulation_parameters.nitsche->number_solids);
+  std::vector<double> heat_flux_on_nitsche_ib_vector(solids.size(), 0);
+
+  // Loops over solids
+  for (unsigned int i_solid = 0; i_solid < solids.size(); ++i_solid)
+    {
+      // Initialize the solid and its particles
+      std::shared_ptr<Particles::ParticleHandler<dim>> &solid_ph =
+        solids[i_solid]->get_solid_particle_handler();
+      double heat_flux_on_nitsche_bc = 0;
+
+      if (this->simulation_parameters.nitsche->nitsche_solids[i_solid]
+            ->enable_heat_bc)
+        {
+          const unsigned int dofs_per_cell = this->fe->dofs_per_cell;
+
+          std::vector<types::global_dof_index> heat_dof_indices(dofs_per_cell);
+          FullMatrix<double>     local_matrix(dofs_per_cell, dofs_per_cell);
+          dealii::Vector<double> local_rhs(dofs_per_cell);
+
+          Function<dim> *solid_temperature =
+            solids[i_solid]->get_solid_temperature();
+
+          // Penalization terms
+          const double beta =
+            this->simulation_parameters.nitsche->nitsche_solids[i_solid]
+              ->beta_heat;
+
+          // Loop over all local particles
+          auto particle = solid_ph->begin();
+          while (particle != solid_ph->end())
+            {
+              local_matrix = 0;
+              local_rhs    = 0;
+
+              const auto &cell = particle->get_surrounding_cell();
+
+              const auto &dh_cell =
+                typename DoFHandler<dim>::cell_iterator(*cell,
+                                                        &this->dof_handler);
+              dh_cell->get_dof_indices(heat_dof_indices);
+
+              const auto pic = solid_ph->particles_in_cell(cell);
+              Assert(pic.begin() == particle, ExcInternalError());
+              for (const auto &p : pic)
+                {
+                  double      temperature = 0;
```

Comment:
Could be good to specify the variable name for future readers (`fluid_temperature` or `temperature_fluid`)